src/tool_getparam.c

```diff
@@ -579,10 +579,9 @@ ParameterError getparameter(const char *flag, /* f or -long-flag */
   (void)clearthis; /* for !HAVE_WRITABLE_ARGV builds */
   *usedarg = FALSE; /* default is that we don't use the arg */
 
-  if(('-' != flag[0]) || ('-' == flag[1])) {
+  if(flag[0] == '-' && flag[1] == '-') {
```

Comment:
This is not the same logic. The previous version also caught long option names without leading dash-dash!

Reply:
> The previous version also caught long option names without leading dash-dash!

I also thought about this initially, but I was unable to trigger that case (invoked like `curl -output foo example.com`; `curl output foo example.com`; ...).

Reply:
The previous logic matched "--verbose" and "verbose" but not "-v", I believe that was the point.

Now you only match "--verbose"